### PR TITLE
kbscan: Disable key repeat for certain SCI keys

### PR DESCRIPTION
**Pull request by crawfxrd**
Resolves: #74

### CODE REVIEW

**Comment by crawfxrd** [comment]
Proprietary firmware on oryp5 has key repeat for display backlight level, but not keyboard backlight level. May want to explicitly list all keys to prevent repeat on instead of all `KT_SCI` keys.

File: src/board/system76/common/kbscan.c [hunk 1]
```diff
@@ -215,6 +215,21 @@ bool kbscan_press(uint16_t key, bool pressed, uint8_t * layer) {
     return true;
 }
 
+static inline bool key_should_repeat(uint16_t key) {
+    switch (key) {
```
Comment (jackpot51) [inline src/board/system76/common/kbscan.c:219]: Also need touchpad toggle

**Comment by crawfxrd** [comment]
Holding the display backlight toggle key while in firmware-setup still flickers the display on and off.

**Comment by jacobgkau** [comment]
Tested this:
- Volume up/down still have key repeat on; it respects GNOME's key repeat setting, so that's fine.
- Display brightness up/down still have key repeat on; they do not respect GNOME's key repeat setting.
- Keyboard backlight brightness up/down still have key repeat on; they do not respect GNOME's key repeat setting.
- All other keys have repeats turned off (within the OS; like Tim noted, within firmware-setup the display on/off repeats very quickly.)

This is an improvement since it stops the display on/off and keyboard backlight on/off from flickering. Are we intending to make the keyboard brightness and display brightness no longer repeat, or are we intending to leave those as-is? (Also wasn't sure if more work would be done to stop the flickering in firmware-setup.)

**Comment by jackpot51** [comment]
I think we want brightness keys to repeat

**Pull request merged**